samples/gltf_viewer.cpp

```diff
@@ -0,0 +1,206 @@
+static std::unique_ptr<MeshAssimp> g_meshSet;
+static const Material* g_material;
+static Entity g_light;
+
+static Config g_config;
+static struct PbrConfig {
+    std::string materialDir;
+    bool clearCoat = false;
+    bool anisotropy = false;
+} g_pbrConfig;
+
+static void printUsage(char* name) {
+    std::string exec_name(Path(name).getName());
+    std::string usage(
+            "gltf_viewer displays gltf models using the filament renderer\n"
+            "Usage:\n"
+            "    gltf_viewer [options] <gltf/glb>\n"
+            "Options:\n"
+            "   --help, -h\n"
+            "       Prints this message\n\n"
+            "   --ibl=<path to cmgen IBL>, -i <path>\n"
+            "       Applies an IBL generated by cmgen's deploy option\n\n"
+            "   --split-view, -v\n"
+            "       Splits the window into 4 views\n\n"
+    );
+    const std::string from("gltf_viewer");
+    for (size_t pos = usage.find(from); pos != std::string::npos; pos = usage.find(from, pos)) {
+        usage.replace(pos, from.length(), exec_name);
+    }
+    std::cout << usage;
+}
+
+static int handleCommandLineArgments(int argc, char* argv[], Config* config) {
+    static constexpr const char* OPTSTR = "hi:v";
+    static const struct option OPTIONS[] = {
+            { "help",           no_argument,       nullptr, 'h' },
+            { "ibl",            required_argument, nullptr, 'i' },
+            { "split-view",     no_argument,       nullptr, 'v' },
+            { 0, 0, 0, 0 }  // termination of the option list
+    };
+    int opt;
+    int option_index = 0;
+    while ((opt = getopt_long(argc, argv, OPTSTR, OPTIONS, &option_index)) >= 0) {
+        std::string arg(optarg ? optarg : "");
+        switch (opt) {
+            default:
+            case 'h':
+                printUsage(argv[0]);
+                exit(0);
+            case 'i':
+                config->iblDirectory = arg;
+                break;
+            case 'v':
+                config->splitView = true;
+                break;
+        }
+    }
+
+    return optind;
+}
+
+static void cleanup(Engine* engine, View* view, Scene* scene) {
+    for (auto& item : g_materialInstances) {
+        auto materialInstance = item.second;
+        engine->destroy(materialInstance);
+    }
+    g_meshSet.reset(nullptr);
+    engine->destroy(g_material);
+
+    EntityManager& em = EntityManager::get();
+    engine->destroy(g_light);
+    em.destroy(g_light);
+}
+
+static void setup(Engine* engine, View* view, Scene* scene) {
+    Path path(g_pbrConfig.materialDir);
+    std::string name(path.getName());
+
+    g_meshSet = std::make_unique<MeshAssimp>(*engine);
+    for (auto& filename : g_filenames) {
+        g_meshSet->addFromFile(filename, g_materialInstances, false);
+    }
+
+    auto& rcm = engine->getRenderableManager();
+    auto& tcm = engine->getTransformManager();
+
+    //Calculate normalize scale so that bounding box is [-1, 1]
+    float maxBound = 0;
```

Comment:
Maybe rename maxBound to maxExtent since it represents a single component of the bounding box size.